packages/mongodb-chatbot-server/src/routes/responses/createResponse.test.ts

```diff
@@ -387,154 +432,186 @@ describe("POST /responses", () => {
         ],
+      };
+      const { response } = await makeClientAndRequest(requestBody);
+      const results = await collectStreamingResponse(response);
+
+      const updatedConversation = await conversations.findByMessageId({
+        messageId: getMessageIdFromResults(results),
       });
+      if (!updatedConversation) {
+        return expect(updatedConversation).not.toBeNull();
+      }
 
-      const createdConversation = await createSpy.mock.results[0].value;
-      const addedMessages = await addMessagesSpy.mock.results[0].value;
+      expect(response.status).toBe(200);
+      testResponses({ requestBody, responses: results });
 
-      expect(response.statusCode).toBe(200);
-      expect(createdConversation.storeMessageContent).toEqual(store);
+      expect(updatedConversation.storeMessageContent).toEqual(store);
 
-      expect(addedMessages[0].role).toEqual("system");
-      expect(addedMessages[1].role).toEqual("system");
+      expect(updatedConversation.messages[0].role).toEqual("system");
+      expect(updatedConversation.messages[0].content).toEqual(functionCallType);
 
-      expect(addedMessages[0].content).toEqual(functionCallType);
-      expect(addedMessages[1].content).toEqual(functionCallOutputType);
+      expect(updatedConversation.messages[1].role).toEqual("system");
+      expect(updatedConversation.messages[1].content).toEqual(
+        functionCallOutputType
+      );
     });
   });
 
   describe("Invalid requests", () => {
     it("Should return 400 with an empty input string", async () => {
-      const response = await makeCreateResponseRequest({
+      const { response } = await makeClientAndRequest({
         input: "",
       });
+      const results = await collectStreamingResponse(response);
 
-      expect(response.statusCode).toBe(400);
-      expect(response.body.error).toEqual(
-        badRequestError(`Path: body.input - ${ERR_MSG.INPUT_STRING}`)
-      );
+      expect(response.status).toBe(200);
+      testInvalidResponses({
+        responses: results,
+        message: `Path: body.input - ${ERR_MSG.INPUT_STRING}`,
+      });
     });
 
     it("Should return 400 with an empty message array", async () => {
-      const response = await makeCreateResponseRequest({
+      const { response } = await makeClientAndRequest({
         input: [],
       });
+      const results = await collectStreamingResponse(response);
 
-      expect(response.statusCode).toBe(400);
-      expect(response.body.error).toEqual(
-        badRequestError(`Path: body.input - ${ERR_MSG.INPUT_ARRAY}`)
-      );
+      expect(response.status).toBe(200);
+      testInvalidResponses({
+        responses: results,
+        message: `Path: body.input - ${ERR_MSG.INPUT_ARRAY}`,
+      });
     });
 
     it("Should return 400 if model is not mongodb-chat-latest", async () => {
-      const response = await makeCreateResponseRequest({
+      const { response } = await makeClientAndRequest({
         model: "gpt-4o-mini",
```

Comment:
nit: can u make this some very obviously not a valid model name, for making the test a bit more readable. smthn like `'not-a-valid-model'`